policy/modules/contrib/knot.te

```diff
@@ -0,0 +1,89 @@
+policy_module(knot, 1.0.0)
+
+########################################
+#
+# Declarations
+#
+
+type knot_t;
+type knot_exec_t;
+init_daemon_domain(knot_t, knot_exec_t)
+
+permissive knot_t;
+
+type knot_conf_t;
+files_type(knot_conf_t)
+
+type knot_var_lib_t;
+files_type(knot_var_lib_t)
+
+type knot_var_run_t;
+files_pid_file(knot_var_run_t)
+
+type knot_tmp_t;
+files_tmp_file(knot_tmp_t)
+
+########################################
+#
+# knot local policy
+#
+allow knot_t self:fifo_file rw_fifo_file_perms;
+allow knot_t self:unix_stream_socket create_stream_socket_perms;
+
+allow knot_t self:tcp_socket create_stream_socket_perms;
+allow knot_t self:udp_socket create_socket_perms;
+allow knot_t self:unix_dgram_socket create_socket_perms;
+
+allow knot_t self:process { setcap };
+allow knot_t self:capability { setpcap };
+
+domain_use_interactive_fds(knot_t)
+
+files_read_etc_files(knot_t)
```

Comment:
Was this permission requested? Domains are already allowed common access to objects with the base_type attribute.